### ISSUE TITLE
Buttons: Add option to format links to look like buttons for accessibility purposes

### ISSUE THREAD

**Issue by ddanier** [primefaces/primevue#3703]
### Describe the feature you would like to see added

Currently PrimeFaces only supports HTML `<button>`'s to look like buttons. This is completely ok when you just think of forms. But if you use buttons to - for example - display a menu or tool bar on a page HTML buttons might not be the best choice. Reason is that `<button>` elements do not contain a `href` attribute to define which URL to navigate to. This means for example handling the `<button>` in a screen reader is much harder.

There might be solutions like putting the button into its own `<form>`, but most developers won't just do that. Also using an `<a href=...>` around the button may be used, but this is just not valid HTML according to standards.

Note this might also be an enhancement for SEO.

### Is your feature request related to a problem?

There is no valid option in PrimeVue to have something looking like a button, but being an `<a href=...>` element. This means you cannot follow accessibility guidelines on how to present UI elements changing the URL of the page - at leats not when needing to have this look like a button.

### Describe the solution you'd like

I think we could take the approach suggested here: https://github.com/primefaces/primevue/pull/3552

This would allow to use the `href` attribute on the `<Button>` component and by doing so switch to using an `<a>` tag. Sadly the PR was closed due to a complete misunderstanding of what has been done here. The PR was there to avoid the validation issue - so the closing reason is just invalid. ;-)

### Describe alternatives you have considered

Although there might be other solutions (like having a `<form>` or defining your own CSS to format the `<a>`) I feel most of those are just bad. Not because they are not valid....but because of the fact that most devs will not use those options. Having things like the need to provide accessible website in the near future (see https://ec.europa.eu/social/main.jsp?catId=1202&intPageId=5581&langId=en) makes this even worse.

In general my feeling is that most UI frameworks tend to just ignore some of the best practices we had established in the web for so many years. I would like to see a framework reiterate on those best practices and support good semantic code.

So I feel the strong need for a good and easy default developers can use.

### Additional context

Accessibility act: https://ec.europa.eu/social/main.jsp?catId=1202&intPageId=5581&langId=en
PR so support `href` on `<Button>`: https://github.com/primefaces/primevue/pull/3552

**Comment by scottrycroft** [comment]
I think putting a `<Button>` inside a `<router-link>` can do what you want. Accessibility via the `<a>` and styling via the `<Button>`
E.g.  this:
```
<router-link :to="{ name: 'home' }"><Button icon="pi pi-search" outlined rounded title="Home" /></router-link>
```

produces this HTML:
```
<a aria-current="page" href="/" class="router-link-active router-link-exact-active">
	<button
		class="p-button p-component p-button-icon-only p-button-rounded p-button-outlined"
		type="button" title="Home" >
		<span class="p-button-icon pi pi-search"></span><span class="p-button-label">&nbsp;</span>
	</button>
</a>

```

**Comment by ddanier** [comment]
Sure, that's...:
* ...what matches the desired behaviour 💪
* ...what your browser might just support (the way you want it to work) 🤞
* ...heck, it's what I actually currently use 😬

Still - and this is important - this is NOT VALID HTML. You are not allowed to mix `<button>` and `<a href>` this way according to HTML standards.

This means this is also:
* ...what you should avoid, as your HTML will not be valid 🧐
* ...what MIGHT change in behaviour as browsers progress (the behaviour is not well defined) 🫠
* ...what MIGHT lead to ugly egde cases you don't want to care about 💣

Background:
* A `<button>` does not allow "interactive" content, while itself is considered an "interactive" element (see https://html.spec.whatwg.org/multipage/form-elements.html#the-button-element)
* A `<a>` may not contain "interactive content", while it is only an "interactive" element if the `href` attribute is set (see https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element)

This means both `<a><button></a>` and `<button><a href></button>` are invalid according to standards.

One example:

```html
<form action="...">
    <a href="...">
        <button>...</button>
    </a>
</form>
```

Should the button now follow the link OR submit the form? 🤯

Meaning: In general, don't use this for production code! There needs to be a better solution to this. 😉

**Comment by scottrycroft** [comment]
Okay, so it's not an accessibility issue then as per the title, but a 'invalid markup'. All current accessibility checkers have no problem with this.
I'm using this in production now, works fine!

**Comment by ddanier** [comment]
It is both. The question is how to get this solved in an accessible way while still ensuring the HTML is valid. Those are two separate quality standards — both are valid and both should be matched.

**Comment by tugcekucukoglu** [comment]
Issue tracker is used for defects only as part of our commitment to quality and continuous improvement in all areas. Enhancements are collected as valuable community feedback and managed internally so moving this enhancement ticket to our internal project management backlog.

**Comment by lateek35** [comment]
@tugcekucukoglu Outputing valid HTML is considered as an enhancements, not a core feature ?

**Comment by robertdeboer** [comment]
I agree. I just ran across this need and the response to this does seem a little interesting, at least to how it is been regulated to be an "Enhancement." While technically it would be I also agree that this is also an issue with rendering correct and valid HTML and so should not just be considered a "nice to have."

**Comment by code3z** [comment]
Just ran into this... it says in the docs I can use <Button as="a" />, but it doesn't work
